cdap-authorization-ldap-role/src/main/java/io/cdap/cdap/security/authorization/ldap/role/RoleAuthorization.java

```diff
@@ -0,0 +1,292 @@
+  public Set<GrantedPermission> listGrants(Principal principal) throws AccessException {
+    throw new AccessException("Method 'listGrants' is not implemented, use external config file instead");
+  }
+
+  PrincipalPermissions getAllPermissions(String username) {
+    Set<String> groups = searcherService.searchGroups(username);
+    return roleProvider.getPrincipalPermissions(groups);
+  }
+
+  private void setSystemUser() {
+    if (!ignoreSystemUser) {
+      try {
+        UserGroupInformation info = UserGroupInformation.getCurrentUser();
+        systemUsername = info.getShortUserName();
+        LOG.info("System user: {}", systemUsername);
+      } catch (IOException e) {
+        ignoreSystemUser = true;
+        LOG.error("Failed to set systemUser, he will be ignored", e);
+      }
+    }
+  }
+
+  private LDAPSearcher createLDAPSearcher(AuthorizationContext context) {
+    Properties properties = context.getExtensionProperties();
+    LOG.info("Searching for LDAP properties in deployment...");
+    LDAPSearchConfig searchConfig = RoleAuthorizationUtil.createSearchConfig(properties);
+
+    if (Objects.isNull(searchConfig.getLookUpBindDN()) && Objects.isNull(searchConfig.getLookUpBindPassword())) {
+      LOG.info("Properties '{}' and '{}' were empty in deployment", RoleAuthorizationConstants.LOOK_UP_BIND_DN,
+               RoleAuthorizationConstants.LOOK_UP_BIND_PASSWORD);
+      LOG.info("Searching for LDAP bind properties in SecureStore...");
+      try {
+        SecureStoreData storeData = context.get(RoleAuthorizationConstants.SYSTEM_NAMESPACE,
+                                                RoleAuthorizationConstants.LDAP_BIND_DN);
+        String bindDn = new String(storeData.get(), StandardCharsets.UTF_8);
+
+        storeData = context.get(RoleAuthorizationConstants.SYSTEM_NAMESPACE, RoleAuthorizationConstants.LDAP_BIND_PASS);
+        String bindPass = new String(storeData.get(), StandardCharsets.UTF_8);
+
+        searchConfig.setLookUpBindDN(bindDn);
+        searchConfig.setLookUpBindPassword(bindPass);
+      } catch (Exception e) {
+        throw new RuntimeException("Failed to get LDAP properties from SecureStore", e);
+      }
+    }
+
+    return new LDAPSearcher(searchConfig);
+  }
+
+  private boolean isVisible(EntityId entityId, PrincipalPermissions principalPermissions) {
+    EntityType entityType = entityId.getEntityType();
+    Permission getPermission = StandardPermission.GET;
+
+    return isAccessible(entityType, entityId, getPermission, principalPermissions);
+  }
+
+  private boolean isAccessible(EntityType entityType, EntityId entityId, Permission permission,
+                               PrincipalPermissions principalPermissions) {
+    // Propagation permissions if entity is not set and propagation is not disabled
+    if (!disablePermissionsPropagation && roleProvider.isEntityUnSet(entityType)) {
+      return RoleAuthorizationUtil.getPropagatedPermission(entityId, permission, principalPermissions)
```

Comment:
I don't think you have to generate a list of propagated permissions. You just have to check for namespace permission

Reply:
We receiving an Optional here and checking if it is present.